### PR TITLE
Only create unit from the columns unit if the to_string worked.

### PR DESCRIPTION
**Pull request by MSeifert04**
Otherwise the unit would be None which would result in an unexpected
(and unnecessary) Exception.

Related: #8897

I'm not sure how much this actually fixes, however since `col.unit` is always `None` in case the `try` block fails - but it seems to make sense only to do the `Unit` call when it succeeded.

### CODE REVIEW

**Comment by MSeifert04** [comment]
Hm, the failures seem real but I don't know what the problem could be...

Probably the most sensible way would be replace the test with a test that only covers the "fixed" case here.

**Comment by mhvk** [comment]
p.s. @MSeifert04 - if you haven't got time to check/write regression check for #8887, I can do it. Just don't want to spend the time if you are already doing it.

p.p.s. Nice find!

**Comment by MSeifert04** [comment]
@mhvk I think (hope) you mixed up some issues. 😄 

I initially thought this would fix #8897, and it kinda did (locally 😅). The problem is really messy because a defined and enabled unit is somehow represented as IrreducibleUnit instead of as `UnrecognizedUnit`. An irreducible unit fails when converting with `to_string('fits')` but roundtrips at least partially - see the first commit here - because the table serialization infrastructure puts it in special comments. The unrecognized unit however works with the `to_string` but isn't properly "read" afterwards.

However the first commit which tried to make a regression test for #8897 failed in some CIs and I haven't had time to investigate it so I reversed the regression test and added one which simply did the isolated test for the non-FITS IrreducibleUnit.

Long story short: If you have time to add the regression test please go ahead. 👍

**Comment by mhvk** [comment]
Yes, mistaken issue number...

I hadn't seen that you had tried the test in the original commit. The python-version dependent failure in travis is very surprising; I can see why you went for the "well, at least this solves something" option!!!

I actually think that is sensible enough that I will just merge this PR, which I feel is obviously correct, and then experiment a bit on the new master (saves me separately downloading your PR...)

**Comment by MSeifert04** [comment]
Thanks @mhvk. :+1:
